.circleci/config.yml

```diff
@@ -176,9 +181,14 @@ commands:
                 environment:
                   RIOT_RUN_RECOMPILE_REQS: "<< pipeline.parameters.riot_run_latest >>"
                 command: |
-                  # Sort the hashes to ensure a consistent ordering/division between each node
-                  riot list --hash-only '<<parameters.pattern>>' | sort | circleci tests split | xargs -n 1 -I {} riot -v run --exitfirst --pass-env -s {} $([[ << pipeline.parameters.coverage >> == false ]] && echo '--no-cov' )
-                  ./scripts/check-diff ".riot/requirements/" "Changes detected after running riot. Consider deleting changed files, running scripts/compile-and-prune-test-requirements and committing the result."
+                  if scripts/needs_testrun.py -v "$CIRCLE_JOB"
```

Comment:
I wonder if we can use an anchor to DRY this? 🤔 

but I am not sure when parameters are replaced... but I think anchors get applied by YAML during loading, so should be basically equivalent....

Reply:
👍  I'd see we keep this PR on point for now, and attempt to DRY this up later, as it doesn't seem to be too trivial